src/components/BackButton/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { Text, TouchableOpacity } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import BackArrow from 'src/assets/images/back-arrow.svg';
+import styles from './styles';
+
+interface BackButtonProps {
+  label: string; // The string prop for the button label
+}
+
+export default function BackButton({ label }: BackButtonProps) {
+  const navigation = useNavigation();
+
+  return (
+    <TouchableOpacity
+      onPress={() => navigation.goBack()}
+      style={styles.backContainer}
+    >
+      <BackArrow />
```

Comment:
I changed some stuff - just moved the back button to be it's own component and take in arguments. Other than that very good work!